tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +144,170 @@ def get_rdm(self):
+
+            for i, val in enumerate(molecule.mo_energies[:-1]):
+                if np.isclose(val, molecule.mo_energies[i+1]):
+                    self.any_close = True
+                    break
+            self.molecule = SecondQuantizedMolecule(xyz=molecule.xyz, q=molecule.q, spin=molecule.spin,
+                                                    solver=IntegralSolverPsi4(),
+                                                    basis=molecule.basis,
+                                                    ecp=molecule.ecp,
+                                                    symmetry=False,
+                                                    uhf=molecule.uhf,
+                                                    frozen_orbitals=molecule.frozen_orbitals)
+        self.basis = molecule.basis
+
+    def simulate(self):
+        """Perform the simulation (energy calculation) for the molecule.
+
+        Returns:
+            float: Total FCI energy.
+        """
+        n_frozen_vir = len(self.molecule.frozen_virtual)
+        n_frozen_occ = len(self.molecule.frozen_occupied)
+        ref = 'rhf' if self.molecule.spin == 0 else 'rohf'
+        self.backend.set_options({'basis': self.basis, 'mcscf_maxiter': 300, 'mcscf_diis_start': 20,
+                                  'opdm': True, 'tpdm': True, 'frozen_docc': [n_frozen_occ], 'frozen_uocc': [n_frozen_vir],
+                                  'qc_module': 'detci', 'fci': True, 'reference': ref})
+
+        # Copy reference wavefunction and swap orbitals to obtain correct active space if necessary
+        wfn = self.backend.core.Wavefunction(self.molecule.solver.mol_nosym, self.molecule.solver.wfn.basisset())
+        wfn.deep_copy(self.molecule.solver.wfn)
+        if n_frozen_occ or n_frozen_vir:
+            mo_order = self.molecule.frozen_occupied + self.molecule.active_occupied + self.molecule.active_virtual + self.molecule.frozen_virtual
+            swap_ops = getswaps(mo_order)
+            for swap_op in swap_ops:
+                wfn.Ca().rotate_columns(0, swap_op[0], swap_op[1], np.deg2rad(90))
+
+        energy, self.ciwfn = self.backend.energy('fci', molecule=self.molecule.solver.mol,
+                                                 basis=self.basis, return_wfn=True,
+                                                 ref_wfn=wfn)
+        return energy
+
+    def get_rdm(self):
+        """Compute the Full CI 1- and 2-particle reduced density matrices.
+
+        Returns:
+            numpy.array: One-particle RDM.
+            numpy.array: Two-particle RDM.
+
+        Raises:
+            RuntimeError: If method "simulate" hasn't been run.
+        """
+
+        # Check if Full CI is performed
+        if self.ciwfn is None:
+            raise RuntimeError("FCISolver: Cannot retrieve RDM. Please run the 'simulate' method first")
+        if self.any_close:
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
```

Comment:
`self.degenerate_orbitals` or `self_degenerate` may be a better choice than `self.is_close` then :)